src/controllers/AuthController.js

```diff
@@ -80,3 +83,64 @@ export const login = async (req, res) => {
     return res.status(500).json({ message: err.message });
   }
 };
+
+// @route POST api/auth/forgot-password
+// @desc Initiates the password recovery process.
+// @access Public
+export const forgotPassword = async (req, res) => {
+  try {
+    // Extract email from the request body
+    const { email } = req.body;
+
+     // Search for a user with the provided email
+    const existingUser = await User.findOne({ email });
+
+    // Check if the user exists with provided email
+    if (!existingUser) {
+      return res.status(400).json({message: 'User not found'});
+    }
+
+    // Generate a random reset token using crypto
+    const resetToken = crypto.randomBytes(20).toString('hex');
+
+    // Assign the reset token and expiration time to the user
+    existingUser.resetPasswordToken = resetToken;
+    existingUser.resetPasswordExpires = Date.now() + 5 * 60 * 1000; // 5 mins
+
+    // Save the updated user information
+    await existingUser.save();
+
+    // Create a transport for sending email using Nodemailer
+    const transporter = nodemailer.createTransport({
+      service: 'gmail',
+      host: 'smtp.gmail.com',
+      port: 587,
+      secure: false,
+      auth: {
+        user: envConfig.mail.user,
+        pass: envConfig.mail.password
+      }
+    });
+
+    // Create the reset password URL, which includes the reset password token.
+    const resetUrl = `${envConfig.clientHost}/reset-password?token=${resetToken}`;
+
+    // Send an email to the user that includes the reset password URL
+    await transporter.sendMail({
+      from: {
+        name: 'Sakura pantry',
+        address: envConfig.mail.user
+      },
+      to: existingUser.email,
+      subject: 'Password Reset',
+      html: `<p>You requested a password reset. Please go to this link to reset your password:<p>\
```

Comment:
Maybe you can make the email content a bit longer and add information about the link expiration. Something like this:

![image](https://github.com/irene2mana-T3A2-2023/SakuraPantryServer/assets/126633021/880226f5-b316-4138-998a-2961e286707f)